internal/pkg/service/stream/sink/type/tablesink/keboola/bridge/job.go

```diff
@@ -73,15 +74,29 @@ func (b *Bridge) CleanJob(ctx context.Context, job model.Job) (err error, delete
 		return err, false
 	}
 
-	token, err := b.schema.Token().ForSink(job.SinkKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
+	existingToken, err := b.schema.Token().ForSink(job.SinkKey).GetOrErr(b.client).Do(ctx).ResultOrErr()
 	if err != nil {
 		b.logger.Warnf(ctx, "cannot get token for sink, already deleted: %s", err.Error())
 		return nil, false
 	}
 
+	// Prepare encryption metadata
+	metadata := cloudencrypt.Metadata{"sink": job.SinkKey.String()}
+
+	// Decrypt token
+	var token keboola.Token
+	if existingToken.EncryptedToken != nil {
+		token, err = b.tokenEncryptor.Decrypt(ctx, existingToken.EncryptedToken, metadata)
+		if err != nil {
+			return err, false
+		}
+	} else {
+		token = *existingToken.Token
+	}
```

Comment:
Couldn't we move this logic into a separate function? It's repeated in several places.